src/utils/logic-app/templateUtils.ts

```diff
@@ -100,7 +101,10 @@ export function generateTemplateParameter(workflow: Workflow) {
 
     for (const key of Object.keys(parameters)) {
         // Rename parameters by prepending `workflows_{workflow}` to avoid conflict of the same parameter name between multiple Logic Apps
-        parameters[`workflows_${workflow.name}_parameters_${key}`] = {value: parameters[key].defaultValue};
+        // Checks whether said parameter already has a value (most likely set from prior deployment), and use its value as defaultValue if exist
+        parameters[`workflows_${workflow.name}_parameters_${key}`] = {
+            value:  (parametersValue as any)[key] && (parametersValue as any)[key].value || parameters[key].defaultValue
```

Comment:
Can you define a type for `parametersValue` instead of using `any`?

Reply:
Will check.

Reply:
As discussed, we will merge this and fix later.